### DISCUSSION TITLE
Documentation on and processing the file.etherpad export file format

### DISCUSSION THREAD

**Discussion by acsr** [ether/etherpad-lite#5148] [Q&A] [answered]
I cannot find by websearch or lookup forums, issues and docs any useful documentation on using the *.etherpad export file format offered in the Etherpad-lite export dialog.

Is there anything I may have overseen? 
Help with deeper pointers or searchterms are welcome...

Reverse engineering and looking up the code and files is the next option, but not preferred.

**Comment by webzwo0i** [comment, marked as answer]
It's a JSON file that contains all database entries related to a pad. (All revisions as they have been stored in the database, atext, all chat entries etc. Plugins can add themselves to a hook that is called when exporting this format, so their entries are included as well.)

**Reply by acsr** [reply]
THX. So is there a documented or best practise way how to import a single exported pad into a different instance of etherpad lite preserving the whole timeline as long as you have an exact clone of the installed code (plugins, module versions) . This should be database agnostic because of the internal abstraction layer.

The current import feature does only allow to import a static version of html loosing still some formatting as you may loss when copying from pad to pad over the browser clipboard. In Chrome e.g. you currently loose inline bold/strong styles. While I managed to paste bold inline style from other sources, this is not reproduceable yet.

So my intention is move entire document creation histories to a different instance (e.g. Archive Instance)
The migration script from dirtyDB to RealDB in bin/migrateDirtyDBtoRealDB.js may contain some hints, but I neeed time to dig deeper.

**Reply by webzwo0i** [reply]
It should be possible by default to import an .etherpad file via browser (it will recognize .etherpad endings by default). The copy&paste bug is already known and will be fixed one day :-D

You can iterate over all pads and import them via curl or wget, too. Something like `curl --form file=@exported.pad.etherpad https://domain:port/p/newpadname/import` should work. The entries in the .etherpad file contain the old pad id, but they will be replaced with the new one during import.

**Reply by acsr** [reply]
THX. Cool for writing this up. I will give it a try and will look forward to feed the results back here or in the wiki.

**Reply by acsr** [reply]
It is quit stupid that I did not do an solely search just for import in the context, looking just for export did not match any useful content. If the content or metadata does not cover the topic you search for, or your experience gives you hints, you are lost on google except asking some real ethernet residents.

Tipp for searching in public documentation is the Algolia docsearch https://docsearch.algolia.com/ a very clever commercial crawler and search UI which is available as a free service for public documentation as a grant to give back to the open source source communities they are building on. I wrote some docs on how to integrate the popular sphinx-doc.org documentations like on readthedoc.org with Algolia docsearch on stackoverflow.  and we use it on https://docs.plone.org/. It clever deals with misspelling and other common search issues. Much better than Google and lower footprint for maintainers instead of running your own ElasticSearch or et al Instance. This is a different shoe for sensible private docs.
